packages/editor/src/components/provider/index.js

```diff
@@ -109,7 +112,8 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 			( select ) => select( editorStore ).getRenderingMode(),
 			[]
 		);
-		const shouldRenderTemplate = !! template && mode !== 'post-only';
+		const shouldRenderTemplate =
+			!! template && mode !== 'post-only' && mode !== 'pattern-only';
```

Comment:
Agree with [Riad's point](https://github.com/WordPress/gutenberg/pull/56827#issuecomment-1856529519) that this kind of thing isn't really scalable.

It seems like `post-only` mode is already mostly appropriate for patterns, but needs a few adjustments for the desired user experience.

For the back button, Riad suggests a rudimentary stack of post types as a history, which sounds reasonable.

Some other aspects, like the dark grey surround around the content can be derived from the post type.